presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -670,7 +670,16 @@ private PlanWithProperties planAndEnforce(
 
             boolean preferenceSatisfied;
             if (pruneExtraHashSymbols) {
-                preferenceSatisfied = result.getHashSymbols().keySet().equals(requiredHashes.getHashes());
+                // Make sure that
```

Comment:
shouldn't be this separated to different commit? Can you add a test for that?

Reply:
@kokosing @sopel39 I don't see any unit tests for `HashGenerationOptimizer` and I cannot use `BaseRuleTest` as it is an old-style rule. What should I do? Is there an example I could follow?